MapboxSearch/offline/src/main/java/com/mapbox/search/offline/OfflineSearchEngine.kt

```diff
@@ -301,6 +301,8 @@ public interface OfflineSearchEngine {
      * Companion object.
      */
     public companion object {
+        private fun buildDatasetName(dataset: String, language: String?): String =
+            language?.let { "$dataset|$it" } ?: dataset
```

Comment:
Consider adding simple validation for the language:
```
fun isValidLanguageCode(code: String): Boolean {
    val locale = Locale.forLanguageTag(code)
    return locale.language.isNotEmpty() && locale.language != "und" // 'und' means undetermined
}
```
or simple regular expression:
```
fun isValidLanguageTag(tag: String): Boolean {
    val languageTagRegex = "^[a-zA-Z]{2,8}(-[a-zA-Z0-9]{2,8})*$".toRegex()
    return languageTagRegex.matches(tag)
}
```